marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -182,8 +187,17 @@ public void awaitCompletion() {
             return output;
         }
 
-        private void processOutput(CallContext callContext){
-            calling: while (true) {
+        private void processOutput() {
+            CallContext callContext = getCallContexts().poll();
+            if(callContext != null) {
+                calling:while (true){
+                    if(!processOutput(callContext))
+                        break calling;
+                }
+            }
+        }
+
+        private Boolean processOutput(CallContext callContext){
```

Comment:
Could the return value be the lower-case primitive boolean instead of the Boolean object?  Is there a reason to autobox?